core/types/block.go

```diff
@@ -378,6 +431,17 @@ func (b *Block) WithBody(transactions []*Transaction, uncles []*Header) *Block {
 	return block
 }
 
+// WithBody2 returns a new block with the given transaction, uncle, and
+// withdrawal contents.
+func (b *Block) WithBody2(transactions []*Transaction, uncles []*Header, withdrawals []*Withdrawal) *Block {
```

Comment:
Convert to separate method `WithWithdrawals` so it can be chained with `WithBody`.